blood/src/main/java/cz/cuni/mff/d3s/blood/dependencyMatrix/DependencyValue.java

```diff
@@ -14,4 +14,13 @@ public void update(long nodesSeen, long nodesTotal) {
         count += nodesSeen;
         totalCount += nodesTotal;
     }
+    
+    @Override
+    public String toString() {
+        return count+":"+totalCount+":"+iterations;
+    }
+    
+    public static String toStringS(DependencyValue dv) {
```

Comment:
I would consider moving this into `utils.StringUtils` as `static <T> String toStringWithDefault(T object, String default)`

Reply:
The intention here was to be able to treat null as an instance of type `DependencyValue` and not in a special way. Your suggestion wouldn't be much more useful than a conditional expression. See usage.

```java
DependencyValue::toStringS
// or
o -> DependencyValue.toStringS(o)
// versus
o -> StringUtils.toStringWithDefault(o, "0:0:0")
// versus
o -> o == null ? "0:0:0" : o.toString()
```

Note also that if we decided to change the string representation to e.g. "X,Y,Z", we would have to change more than just `DependencyValue.java`.

I agree that the name of the method is not spectacular ;)

Reply:
I think the example in the middle is the best. At first sight it's clear what it does. The conditional expression bellow requires more effort when reading and the top one is unclear with it's name. And I didn't came up with any better name, so instead I suggested generalizing it. :wink:

Reply:
I see it as an abstraction: it's not a reference that can be null, but a structure that can only contain nulls and that is always safe to be stringified.
Requiring to make sure it's not null is leaking implementation.

Reply:
* was resolved as a side effect of refactoring *